pydrive2/test/test_oauth.py

```diff
@@ -125,6 +126,27 @@ def test_09_SaveLoadCredentialsUsesDefaultStorage(mocker):
     assert spy.call_count == 0
 
 
+def test_10_LocalWebserverAuthFromSavedCredentialsDictionary():
+    creds_dict = {"creds": os.getenv(GDRIVE_USER_CREDENTIALS_DATA)}
+    settings = {
+        "client_config_backend": "settings",
+        "client_config": {
+            "client_id": "47794215776-cd9ssb6a4vv5otkq6n0iadpgc4efgjb1.apps.googleusercontent.com",  # noqa: E501
+            "client_secret": "i2gerGA7uBjZbR08HqSOSt9Z",
+        },
+        "oauth_scope": ["https://www.googleapis.com/auth/drive"],
+        "save_credentials": True,
+        "save_credentials_backend": "dictionary",
+        "save_credentials_dict": creds_dict,
+        "save_credentials_key": "creds",
+    }
+    ga = GoogleAuth(settings=settings)
+    # Shouldn't hang to wait for manual auth, as creds_dict already has the creds
+    ga.LocalWebserverAuth()
```

Comment:
If I understand correctly the creds data is for a service account, hence why this fails. Will get back to this in the morning.